budgetApp1/budgetApp1/AccountTransactionsViewController.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  AccountTransactionsViewController.swift
+//  budgetApp1
+//
+//  Created by Lehlohonolo Nkadimeng on 2024/05/12.
+//
+
+import UIKit
+
+class AccountTransactionsViewController: UIViewController {
+
+    // MARK: - UISpecs Dependency
+    private let uiSpecs = UISpecs()
+    
+    // MARK: - Variables
+    @IBOutlet private weak var tableView: UITableView!
+    @IBOutlet private weak var segmentedControl: UISegmentedControl!
+    @IBOutlet private weak var accountNameLabel: UILabel!
+    private lazy var viewModel = AccountTransactionsViewModel(repository: TransactionRepository(), delegate: self)
+    var selectedAccount: Account?
+
+    // MARK: - functions
+    @IBAction private func segmentedChange(_ sender: AnyObject) {
+        viewModel.filterTransactions(segmentIndex: sender.selectedSegmentIndex)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        viewModel.fetchTransactionsForAccount(accountID: selectedAccount?.id ?? "")
+        setupTableView()
+        setUpSegmentedControl()
+        setUpAccountDetailsLayout()
+    }
+    
+    private func setupTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(TransactionTableViewCell.nib(), forCellReuseIdentifier: NibIdentifiers.TransactionViewCellIdentifier)
+        tableView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 1.0, right: 0)
+    }
+    
+    private func setUpSegmentedControl() {
+        segmentedControl.layer.borderColor = uiSpecs.accentColour.cgColor
+        segmentedControl.layer.borderWidth = 2.0
+    }
+    
+    private func setUpAccountDetailsLayout() {
+        accountNameLabel.text = selectedAccount?.name
+        accountNameLabel.textColor = uiSpecs.primaryColourOne
+    }
+}
+
+// MARK: - TableView Delegate
+extension AccountTransactionsViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        60.0
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if segmentedControl.selectedSegmentIndex == 0 {
+            viewModel.expenses?.count ?? 0
+        } else {
+            viewModel.income?.count ?? 0
+        }
```

Comment:
Please use a ternary operator here instead of if else

```suggestion
        segmentedControl.selectedSegmentIndex == 0 
        ? viewModel.expenses?.count ?? 0
        : viewModel.income?.count ?? 0
```

Reply:
Since I now have 3 indices in my segmentedControl, I use switch statements